airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/ConfiguredSyncMode.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.read
+
+import io.airbyte.protocol.models.v0.SyncMode
+
+/**
+ * [ConfiguredSyncMode] is equivalent to [SyncMode].
+ *
+ * This exists to avoid coupling the Bulk CDK code too closely to the Airbyte Protocol objects.
+ */
+enum class ConfiguredSyncMode {
```

Comment:
why no keep `SyncMode`? `ConfigureSyncMode` sounds like there's an `unconfigured` version.
Another scheme I used in a previous life was to prefix all the object types, like in `AirbyteSyncMode`. We do have some constructs (most notably `Stream` that already exist in java and could use a differenciator

Reply:
If I kept it, it would be the only remaining protocol object which isn't being used only at the seams.
If a protocol upgrade brings with it a new enum value, then we're in for a world of pain!

> ConfigureSyncMode sounds like there's an unconfigured version.

I'm not so sure about that.

Reply:
what I meant by "keep" was "why not reuse the same name". I'm all for not using the original protocol objects!